CONTRIBUTING.rst

```diff
@@ -0,0 +1,177 @@
+   git clone --branch development https://github.com/AMReX-Codes/amrex.git
+
+   # Clone your fork on your local computer. You can get this address on your fork's Github page.
+   git clone https://github.com/<myGithubUsername>/WarpX.git
+   cd warpx
+   # Keep track of the main WarpX repo, to remain up-to-date.
+   git remote add upstream https://github.com/ECP-WarpX/WarpX.git
+
+Now you are free to play with your fork (for additional information, you can visit the
+`Github fork help page <https://help.github.com/en/articles/fork-a-repo>`_).
+
+.. note::
+
+   You do not have to re-do the setup above every time.
+   nstead, in the future, you need to update the ``master`` branch on your fork with
+
+   .. code-block:: sh
+
+      git checkout master
+      git pull upstream master
+
+Make sure you are on WarpX ``master`` branch with
+
+.. code-block:: sh
+
+   git checkout master
+
+in the WarpX directory.
+
+Create a branch ``<branch_name>`` (the branch name should reflect the piece of code you want to add, like ``fix_spectral_solver``) with
+
+.. code-block:: sh
+
+   git checkout -b <branch_name>
+
+and do the coding you want.
+It is probably a good time to look at the `AMReX documentation <https://amrex-codes.github.io/amrex/docs_html/>`_ and at the `AMReX Doxygen reference <https://ccse.lbl.gov/pub/AMReX_Docs/index.html>`_.
+Add the files you work on to the git staging area with
+
+.. code-block:: sh
+
+   git add <file_I_created> <and_file_I_modified>
+
+Commit & push your changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Periodically commit your changes with
+
+.. code-block:: sh
+
+   git commit -m "This is a 50-char description to explain my work"
+
+The commit message (between quotation marks) is super important in order to follow the developments and identify bugs.
+
+For the moment, commits are on your local repo only.
+You can push them to your fork with
+
+.. code-block:: sh
+
+   git push -u origin <branch_name>
+
+If you want to synchronize your branch with the ``master`` branch (this is useful when the ``master`` branch is being modified while you are working on ``<branch_name>``), you can use
+
+.. code-block:: sh
+
+   git pull upstream master
+
+and fix any conflict that may occur.
+
+Submit a Pull Request
+^^^^^^^^^^^^^^^^^^^^^
+
+A Pull Request (PR) is the way to efficiently visualize the changes you made and to propose your new feature/improvement/fix to the WarpX project.
+Right after you push changes, a banner should appear on the Github page of your fork, with your ``<branch_name>``.
+
+- Click on the ``compare & pull request`` button to prepare your PR.
+- It is time to communicate your changes: write a title and a description for your PR.
+  People who review your PR are happy to know
+
+  * what feature/fix you propose, and why
+  * how you made it (created a new class than inherits from...)
+  * and anything relevant to your PR (performance tests, images, *etc.*)
```

Comment:
```suggestion
  * what feature/fix you propose, and why
  * how you made it (files changed fro what reason, created a new class than inherits from...)
  * how you tested it and what was the output you got
  * and anything relevant to your PR (performance tests, images, *etc.*)
```

Reply:
would not recommend that one has to explain why one touched each file. people might start explaining "first I changed the header because that's were function signatures go." which is a bit too verbose.
maybe `added new/edited files, create a new class that ...` is probably good